### PR TITLE
UX: composer use BEM + small tweaks to enabled/disabled styling

### PR DESCRIPTION
**Pull request by chapoi**
* using BEM notation
* making animation linear instead of default ease
* small tweaks to composer state (disabled/send-disabled/send-enabled)
* fixing bug with disabled composer on mobile
image

### CODE REVIEW

File: plugins/chat/spec/system/page_objects/chat/chat_channel.rb [hunk 1]
```diff
@@ -18,7 +18,7 @@ def click_composer
       end
 
       def click_send_message
-        find(".chat-composer--send-enabled .chat-composer__send-btn").click
+        find(".chat-composer is-enabled .chat-composer__send-btn").click
```
Comment (jjaffeux) [inline plugins/chat/spec/system/page_objects/chat/chat_channel.rb:21]: this needs to be `.chat-composer.is-enabled`

File: plugins/chat/spec/system/page_objects/chat/chat_thread.rb [hunk 2]
```diff
@@ -46,7 +46,7 @@ def send_message(id, text = nil)
 
       def click_send_message(id)
         find(thread_selector_by_id(id)).find(
-          ".chat-composer--send-enabled .chat-composer__send-btn",
+          ".chat-composer is-enabled .chat-composer__send-btn",
```
Comment (jjaffeux) [inline plugins/chat/spec/system/page_objects/chat/chat_thread.rb:49]: this needs to be `.chat-composer.is-enabled`

File: plugins/chat/assets/javascripts/discourse/components/chat-composer.hbs [hunk 3]
```diff
@@ -17,14 +17,10 @@
...
-      (if this.pane.sending "chat-composer--sending")
-      (if
-        this.sendEnabled
-        "chat-composer--send-enabled"
-        "chat-composer--send-disabled"
-      )
-      (if this.disabled "chat-composer--disabled")
+      (if this.isFocused "is-focused")
+      (if this.pane.sending "is-sending")
+      (if this.sendEnabled "is-enabled" "is-disabled")
```
[the remark this answers is not in the document]
Comment (jjaffeux) [reply plugins/chat/assets/javascripts/discourse/components/chat-composer.hbs:22]: yes indeed

[the remark this answers is not in the document]
Comment (jjaffeux) [reply plugins/chat/assets/javascripts/discourse/components/chat-composer.hbs:22]: AS in you have the right to use the composer but you didnt type anything so send btn is not enabled

[the remark this answers is not in the document]
Comment (chapoi) [reply plugins/chat/assets/javascripts/discourse/components/chat-composer.hbs:22]: k ty will change it

**Review by jjaffeux — APPROVED** [review]

**Pull request merged**